library/modules/Items.cpp

```diff
@@ -720,13 +720,13 @@ df::coord Items::getPosition(df::item *item)
     return item->pos;
 }
 
-static char quality_table[] = { 0, '-', '+', '*', '=', '@' };
+static int quality_table[] = { 0, 45, 43, 42, 240, 15 };
```

Comment:
What's the reasoning for changing these to `int`s? IMO they decrease readability.

If it must contain integers due to the `240` being out of range for a signed int8, you could use `static_cast<int>('-')`, etc.

Reply:
sorry, the reasoning was "i have no idea what i'm doing, but this seems to work!" :flushed:

all i knew was that the following didn't work:
(https://github.com/TaxiService/dfhack/actions/runs/4576523769)
```
[-Werror=multichar]
  723 | static char quality_table[] = { 0, '-', '+', '*', '≡', '☼' };
      |                                                   ^~~
/home/runner/work/dfhack/dfhack/library/modules/Items.cpp:723:58: error: multi-character character constant [-Werror=multichar]
  723 | static char quality_table[] = { 0, '-', '+', '*', '≡', '☼' };
      |                                                        ^~~
/home/runner/work/dfhack/dfhack/library/modules/Items.cpp:723:64: error: narrowing conversion of ‘14846369’ from ‘int’ to ‘char’ [-Wnarrowing]
  723 | static char quality_table[] = { 0, '-', '+', '*', '≡', '☼' };
      |                                                            ^
/home/runner/work/dfhack/dfhack/library/modules/Items.cpp:723:64: error: narrowing conversion of ‘14850236’ from ‘int’ to ‘char’ [-Wnarrowing]
```
and then myk suggested to use this "static_cast<char>(999)" thing; so i cobbled something together and it miraculously worked. 

but yeah he too had some concerns about readability. I'll see what i can do to make this pr more sensible. 💪

Reply:
Is there a reason why we can't just use `\xF0` for '≡' and `\x0F` for '☼'?

Reply:
the only reason is the same of my previous post. (idk what im doing + it seems to work!)

how would the readability compare with my latest commit? https://github.com/DFHack/dfhack/pull/3137/commits/3a6893de53ee3994aa68641dd8b49a423ff617ab

Reply:
I like that new commit.

The reason pasting the literals in didn't work (https://github.com/DFHack/dfhack/pull/3137#discussion_r1155103028) is that a `char` in C is only 8 bits, and the Unicode characters you pasted in require more than 8 bits to represent (that's what the `14846369` means in one of your error messages - it's the numerical representation of your character in Unicode, which is much bigger than the allowed maximum of 255).